src/providers/oauth-provider.js

```diff
@@ -71,6 +49,37 @@ function OAuthProvider() {
     }
 
     return config;
+}
+
+
+/**
+ * OAuth provider.
+ */
+
+function OAuthProvider() {
+  var defaultConfig;
+
+  /**
+   * Configure.
+   *
+   * @param {object} params - An `object` of params to extend.
+   */
+
+  this.configure = function(params) {
+    // Can only be configured once.
+    if (defaultConfig) {
+      throw new Error('Already configured.');
+    }
+
+    // Check if is an `object`.
+    if (!(params instanceof Object)) {
+      throw new TypeError('Invalid argument: `config` must be an `Object`.');
+    }
+
+    // Extend default configuration.
+    defaultConfig = angular.extend({}, defaults, params);
```

Comment:
while I'm currently applying your remarqs, it appear to me that using a 3rd argument to override specifically config is not very handy 
``` js
 OAuth.getAccessToken({
  username: 'foo',
  password: 'bar'
}, {}, config);
``` 
that's why I iniitially reuse the options parameter 
What do you think ?